service/frontend/workflow_handler.go

```diff
@@ -4691,15 +4692,15 @@ func (wh *WorkflowHandler) validateRetryPolicy(namespaceName namespace.Name, ret
 func (wh *WorkflowHandler) validateStartWorkflowTimeouts(
 	request *workflowservice.StartWorkflowExecutionRequest,
 ) error {
-	if timestamp.DurationValue(request.GetWorkflowExecutionTimeout()) < 0 {
+	if !timer.ValidateTimer(request.GetWorkflowExecutionTimeout()) {
 		return errInvalidWorkflowExecutionTimeoutSeconds
 	}
 
-	if timestamp.DurationValue(request.GetWorkflowRunTimeout()) < 0 {
+	if !timer.ValidateTimer(request.GetWorkflowRunTimeout()) {
 		return errInvalidWorkflowRunTimeoutSeconds
 	}
 
-	if timestamp.DurationValue(request.GetWorkflowTaskTimeout()) < 0 {
+	if !timer.ValidateTimer(request.GetWorkflowTaskTimeout()) {
 		return errInvalidWorkflowTaskTimeoutSeconds
```

Comment:
should the error message return more information indicating what the min/maximum is?